### ISSUE TITLE
Gzip flag overrides output file flag

### ISSUE THREAD

**Issue by lazear** [compomics/ThermoRawFileParser#148]
Hi,

Using a combination of `-g` and `-b` flags produces an unexpected result.

`.\ThermoRawFileParser.exe -N -f 1 -g -i="test.raw" -b="test.mzML.gz"` creates a file named "test.mzML.gz.gzip"

I would also like to request, perhaps, that the default extension for using the `-g` flag be changed to `.gz`, which is the standard extension from gzipped files (for instance, gunzip doesn't recognize `.gzip`....)

**Comment by caetera** [comment]
Hi @lazear,

thank you for reporting this issue. Do I understand correctly, that "less surprising" result will be to use provided filename as it is (in layman terms - "user knows better")?
For example, `.\ThermoRawFileParser.exe -g -i="test.raw" -b="test.mzML"` should generate `test.mzML` that is in fact a gzipped one.

**Comment by edeutsch** [comment]
Sorry to barge in on the convo, but I think the least surprising is if:
.\ThermoRawFileParser.exe -g -i="test.raw" -b="test.mzML"
generates test.mzML.gz

and:
.\ThermoRawFileParser.exe -g -i="test.raw" -b="test.mzML.gz"
generates test.mzML.gz

(like "Save As...." applications with format selection do)

**Comment by lazear** [comment]
> thank you for reporting this issue. Do I understand correctly, that "less surprising" result will be to use provided filename as it is (in layman terms - "user knows better")? For example, `.\ThermoRawFileParser.exe -g -i="test.raw" -b="test.mzML"` should generate `test.mzML` that is in fact a gzipped one.

I would expect that the generated mzML file name to exactly match the file name passed by the `-b` file ("user knows better"), but I think @edeutsch's suggestion (where the gz file extension is appended if not present) is probably the better approach for most users/applications (which generally rely on file extensions)

**Comment by caetera** [comment]
Thank you both for your input. I will stick to "add the extension, if it is not present". I guess, for unification, the same behavior should be added to other file formats, i.e. add `.mzML` to all mzML files, if it is not present and so on.

**Comment by caetera** [comment]
I have implemented the changes and added a test with the following table of test cases (with as strange user inputs as I can come with). Please, let me know, if there is any case that I might forgot to include. (I guess the table is rather self-explanatory)

Format | Gzip | UserInput | FileOutput
-- | -- | -- | --
MGF | FALSE | rawfile | rawfile.mgf
MGF | FALSE | rawfile. | rawfile.mgf
MGF | FALSE | rawfile.mgf | rawfile.mgf
MGF | FALSE | rawfile.MGF | rawfile.mgf
MGF | FALSE | rawfile.. | rawfile.mgf
MGF | TRUE | rawfile.mgf.gz | rawfile.mgf.gz
MGF | TRUE | rawfile | rawfile.mgf.gz
MGF | TRUE | rawfile.. | rawfile.mgf.gz
MGF | TRUE | rawfile.mgf.GZ | rawfile.mgf.gz
MGF | TRUE | rawfile.mgf | rawfile.mgf.gz
MGF | TRUE | rawfile.MGF. | rawfile.mgf.gz
mzML | FALSE | rawfile | rawfile.mzML
mzML | FALSE | rawfile. | rawfile.mzML
mzML | FALSE | rawfile.mgf | rawfile.mgf.mzML
mzML | FALSE | rawfile.MzML | rawfile.mzML
mzML | FALSE | rawfile.. | rawfile.mzML
mzML | TRUE | rawfile.mzML.gz | rawfile.mzML.gz
mzML | TRUE | rawfile | rawfile.mzML.gz
mzML | TRUE | rawfile.. | rawfile.mzML.gz
mzML | TRUE | rawfile.mzml.GZ | rawfile.mzML.gz
mzML | TRUE | rawfile.mzML | rawfile.mzML.gz
mzML | TRUE | rawfile.MZml. | rawfile.mzML.gz

**Comment by lazear** [comment]
That looks good to me, thank you for tackling this @caetera! I think it would also be good to update the documentation & CLI flag help to clarify that the flag only effects the filename, and not the full file path.

**Comment by caetera** [comment]
Closed in version 1.4.2
